mender-update/deployments/platform/boost_log/deployments.cpp

```diff
@@ -0,0 +1,197 @@
+// Copyright 2023 Northern.tech AS
+//
+//    Licensed under the Apache License, Version 2.0 (the "License");
+//    you may not use this file except in compliance with the License.
+//    You may obtain a copy of the License at
+//
+//        http://www.apache.org/licenses/LICENSE-2.0
+//
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+
+#include <mender-update/deployments.hpp>
+
+#include <boost/date_time/posix_time/posix_time.hpp>
+#include <boost/log/attributes.hpp>
+#include <boost/log/common.hpp>
+#include <boost/log/sinks.hpp>
+#include <boost/smart_ptr/shared_ptr.hpp>
+
+#include <filesystem>
+#include <string>
+
+#include <common/error.hpp>
+#include <common/io.hpp>
+#include <common/json.hpp>
+#include <common/log.hpp>
+#include <common/path.hpp>
+
+namespace mender {
+namespace update {
+namespace deployments {
+
+using namespace std;
+
+namespace logging = boost::log;
+namespace expr = boost::log::expressions;
+namespace sinks = boost::log::sinks;
+
+namespace fs = std::filesystem;
+
+namespace error = mender::common::error;
+namespace io = mender::common::io;
+namespace json = mender::common::json;
+namespace mlog = mender::common::log;
+namespace path = mender::common::path;
+
+static void JsonLogFormatter(logging::record_view const &rec, logging::formatting_ostream &strm) {
```

Comment:
They are not supported by https://docs.mender.io/api/#device-api-deployments-schemas-deploymentlog AFAICT

Reply:
And `level` is actually `required`... Damn

Reply:
Yeah, but it's probably better to have an entry without it in the file and then report it being incomplete somewhere else because the actual message is the most important bit here.